R/toolISOhistorical.R

```diff
@@ -115,56 +116,253 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
+                                 fill = 0)
+        z <- mbind(z, missingNew)
+
+        vcat(1, "Data was newly created and given equal weighting for the following new countries in dim 1.1: ",
+             getItems(missingNew, dim = mainDim))
+      }
+
+      # create transformation matrix
+      # time span where the data need to be adjusted
+      subTime <- getYears(z[, seq_len(which(getYears(z) == tr$fromY)), ])
+      # disaggregation of countries
+
+      if (length(tr$fromISO) == 1) {
+
+        if (is.null(additional_weight)) {
+
+          weight <- setYears(z[setNames(list(c(tr$toISO)), mainDimName), tr$toY, ], NULL)
+          # mapping needs to be subset to the exact combinations present in the weight for toolAggregate
+          # if weight does not contain some of the combinations in m add it
+          addR <- setdiff(getItems(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ], dim = secdDim),
+                          getItems(weight, dim = secdDim))
+          if (length(addR) > 0) {
+            if (mainDim == 1.1) {
+              addR <-  as.vector(outer(getItems(weight, dim = mainDim), addR, paste, sep = "."))
+            } else if (mainDim == 1.2) {
+              addR <- as.vector(outer(addR, getItems(weight, dim = mainDim), paste, sep = "."))
+            }
+}
+          # sometimes individual bilateral combination may be missing, leading to weights that don't sum to 1
+          if (mainDim == 1.1) {
+            addInd <- setdiff(as.vector(outer(
+              tr$toISO,
+              getItems(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ], dim = secdDim),
+              paste, sep = ".")),
+              getItems(weight, dim = 1))
+          } else if (mainDim == 1.2) {
+            addInd <- setdiff(as.vector(outer(
+              getItems(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ], dim = secdDim),
+              tr$toISO, paste, sep = ".")),
+              getItems(weight, dim = 1))
+          }
+          if (length(addInd) > 0) {
+            addR <- unique(c(addR, addInd))
+          }
+          if (length(addR) > 0) {
+            addMiss <- new.magpie(cells_and_regions =  addR,
+                                  years = getItems(weight, dim = 2),
+                                  names = getItems(weight, dim = 3),
+                                  fill = 0)
+            weight <- mbind(weight, addMiss)
+          }
+
+
+          if (anyNA(weight)) {
+            weight[is.na(weight)] <- 0
+            vcat(0, "Weight 1 in toolISOhistorical contained NAs. Set NAs to 0!")
+          }
+        } else {
+          stop("additional_weight is not supported for bilateral data.")
         }
-        weight <- additional_weight[a$toISO, , ]
+
+        bilatMapping <- mapping # mapping needs to be made bilateral
+        if (mainDim == 1.1) {
+bilatMapping <- data.frame(toISO = as.vector(outer(bilatMapping$toISO, getItems(z, dim = secdDim), paste, sep = ".")),
```

Comment:
Please fix the indentation here. If the line is too long with proper indentation I'd suggest to pull out the toISO part: Assign a variable `toISO` in the line before and just set `toISO = toISO` when creating the data.frame.
In case you want to use the RStudio auto-indentation fix again, please run `lucode2::autoFormat()` afterwards to remove trailing whitespace which RStudio introduces when fixing indentation.